ducktape/cluster/json.py

```diff
@@ -31,7 +32,7 @@ class JsonCluster(Cluster):
     """An implementation of Cluster that uses static settings specified in a cluster file or json-serializeable dict
     """
 
-    def __init__(self, cluster_json=None, *args, **kwargs):
+    def __init__(self, cluster_json: Optional[Union[dict, str]]=None, *args, **kwargs):
```

Comment:
we should upstream these changes and it looks like they support back to py3.6. do our typings (and in particular the built-in typing ie. dict vs Dict etc...) work in 3.6?

Reply:
nope.. 3.6 uses the uppercase `typing` definitions, IIRC. I agree with upstreaming. I think I'd propose holding of on a final decision until I try the "just do it in terraform" approach. I think the things I learned here will be helpful even if we end up abandoning this approach.